zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/authorization/DbRoleState.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Camunda License 1.0. You may not use this file
+ * except in compliance with the Camunda License 1.0.
+ */
+package io.camunda.zeebe.engine.state.authorization;
+
+import io.camunda.zeebe.db.ColumnFamily;
+import io.camunda.zeebe.db.TransactionContext;
+import io.camunda.zeebe.db.ZeebeDb;
+import io.camunda.zeebe.db.impl.DbCompositeKey;
+import io.camunda.zeebe.db.impl.DbForeignKey;
+import io.camunda.zeebe.db.impl.DbLong;
+import io.camunda.zeebe.db.impl.DbString;
+import io.camunda.zeebe.engine.state.mutable.MutableRoleState;
+import io.camunda.zeebe.protocol.ZbColumnFamilies;
+
+public class DbRoleState implements MutableRoleState {
+
+  private final DbLong roleKey;
+  private final ColumnFamily<DbLong, PersistedRole> roleColumnFamily;
+
+  private final DbForeignKey<DbLong> fkRoleKey;
+  private final DbLong entityKey;
+  private final DbCompositeKey<DbForeignKey<DbLong>, DbLong> fkRoleKeyAndEntityKey;
+  private final ColumnFamily<DbCompositeKey<DbForeignKey<DbLong>, DbLong>, EntityTypeValue>
+      entityTypeByRoleColumnFamily;
+
+  private final DbString roleName;
+  private final ColumnFamily<DbString, DbForeignKey<DbLong>> roleByNameColumnFamily;
+
+  public DbRoleState(
+      final ZeebeDb<ZbColumnFamilies> zeebeDb, final TransactionContext transactionContext) {
+    roleKey = new DbLong();
+    roleColumnFamily =
+        zeebeDb.createColumnFamily(
+            ZbColumnFamilies.ROLE, transactionContext, roleKey, new PersistedRole());
```

Comment:
```suggestion
            ZbColumnFamilies.ROLE, transactionContext, roleKey, new PersistedRole());
```

❌ We should make the `new PersistedRole()` a field. If we don't we need to construct a new object when we want to create it. Generally creating these objects is expensive.